lib/common/database.dart

```diff
@@ -28,40 +28,59 @@ class DatabaseService {
   }
 
   /// Assign UID. This must be ran before any other database function is called else it will crash
-  /// 
   /// takes the string ID
   initUID(String uid) {
     userid = uid;
   }
 
-  Future<DocumentSnapshot<Map<String, dynamic>>> getEvents(String eventID) {
-    return users
-        .doc(userid)
-        .collection("events")
-        .doc(eventID)
-        .get(); // turn this into a map of eventID to event objects?
+  Future<Event> getEvent(String eventID) async {
+    try {
+      DocumentSnapshot<Map<String, dynamic>> eventDocument =
+          await users.doc(userid).collection('events').doc(eventID).get();
+      if (eventDocument.exists && eventDocument.data() != null) {
+        return Event.fromMap(eventDocument.data() ?? {"getEvent Error": 1},
+            id: eventDocument.id);
+      }
+    } catch (e) {
+      rethrow;
+    }
+    throw Exception("Event not found");
   }
 
-  Future<QuerySnapshot<Map<String, dynamic>>> getAllEvents() {
-    return users
+  Future<List<Event>> getAllEvents() async {
+    List<Event> allEvents = [];
+    final eventDocs = await users.doc(userid).collection("events").get();
+
+    for (var doc in eventDocs.docs) {
+      allEvents.add(Event.fromMap(doc.data(), id: doc.id));
+    }
+
+    return allEvents;
+  }
+
+  /// Put a new event in database. Error if already exists
+  Future<void> addEvent(Event event) async {
+    var doc = await users.doc(userid).collection("events").doc(event.id).get();
+    if (doc.exists) {
+      throw Future.error("Event ID already exists!");
+    }
+    return await users
         .doc(userid)
         .collection("events")
-        .get();
+        .doc(event.id)
```

Comment:
This forces the user to deal with creating unique event IDs. The original function was made so that the user doesn't have to worry about that. Is there a way to either enforce this or deal with it ourselves? Does the event class currently create unique event IDs?

Reply:
The superclass handles id generation in the constructor now. The undertaking constructor has allows us to pass in an id if we are constructing an event obj that comes from the database, or we can leave it as null and it will autogenerate from datetime.now.millisecondsSinceEpoch.toString

Reply:
sounds good